src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -80,149 +97,168 @@ Item {
+                            previousText = text
                         }
-                        previousText = text
+                    }
+                    StatusBaseText {
+                        id: placeholder
+                        visible: (edit.text.length === 0)
+                        anchors.left: parent.left
+                        anchors.right: parent.right
+                        anchors.rightMargin: statusBaseInput.rightPadding
+                        anchors.verticalCenter: parent.verticalCenter
+                        font.pixelSize: 15
+                        elide: StatusBaseText.ElideRight
+                        font.family: Theme.palette.baseFont.name
+                        color: statusBaseInput.enabled ? Theme.palette.baseColor1 :
+                                                        Theme.palette.directColor6
                     }
                 }
-                StatusBaseText {
-                    id: placeholder
-                    visible: (edit.text.length === 0)
-                    anchors.left: parent.left
-                    anchors.right: parent.right
-                    anchors.rightMargin: statusBaseInput.rightPadding
-                    anchors.verticalCenter: parent.verticalCenter
-                    font.pixelSize: 15
-                    elide: StatusBaseText.ElideRight
-                    font.family: Theme.palette.baseFont.name
-                    color: statusBaseInput.enabled ? Theme.palette.baseColor1 :
-                                                     Theme.palette.directColor6
-                }
-            }
 
-        } // Flickable
-        MouseArea {
-            id: sensor
-            enabled: !edit.activeFocus
-            hoverEnabled: true
-            anchors.fill: parent
-            cursorShape: Qt.IBeamCursor
-            onClicked: edit.forceActiveFocus()
-        }
+            } // Flickable
 
+            Item {
+                id: statusBaseInputComponentSlot
+                anchors.right: parent.right
+                anchors.rightMargin: 12
+                width: childrenRect.width
+                height: childrenRect.height
+                anchors.verticalCenter: parent.verticalCenter
+            }
+        }
     } // Rectangle
 
-    StatusFlatRoundButton {
-        id: clearButton
-        visible: edit.text.length != 0 &&
-                 statusBaseInput.clearable &&
-                 !statusBaseInput.multiline &&
-                 edit.activeFocus
-        anchors.right: parent.right
-        anchors.rightMargin: 8
-        anchors.verticalCenter: parent.verticalCenter
-        type: StatusFlatRoundButton.Type.Secondary
-        width: 24
-        height: 24
-        icon.name: "clear"
-        icon.width: 16
-        icon.height: 16
-        icon.color: Theme.palette.baseColor1
-        onClicked: {
-            edit.clear()
+    Loader {
+        id: clearButtonLoader
+        sourceComponent: StatusFlatRoundButton {
```

Comment:
This button in the expected position? I ask bc i don't see anchors or layout

Reply:
Yes, this is because it gets loaded into an `Item` that has the correct possible. Check `onClearableChanged` handler